dht/range_streamer.cc

```diff
@@ -222,10 +222,16 @@ range_streamer::get_all_ranges_with_strict_sources_for(const sstring& keyspace_n
 bool range_streamer::use_strict_sources_for_ranges(const sstring& keyspace_name) {
     auto& ks = _db.local().find_keyspace(keyspace_name);
     auto& strat = ks.get_replication_strategy();
-    return !_db.local().is_replacing()
+    auto rf = strat.get_replication_factor();
+    auto nr_nodes_in_ring = get_token_metadata().get_all_endpoints().size();
+    // Use strict when number of nodes in the ring is equal or more than RF
+    auto strict = !_db.local().is_replacing()
            && _db.local().get_config().consistent_rangemovement()
            && !_tokens.empty()
-           && get_token_metadata().get_all_endpoints().size() != strat.get_replication_factor();
+           && nr_nodes_in_ring >= rf;
```

Comment:
I didn't understand why the "!=" changed to ">=" (and it's not even >, it's >=?)

Reply:
The != comes from C* which is wrong. It is considering only the case where there are more nodes in the cluster than rf. It is >= because, say we have 3 nodes in the cluster, and the rf is 3, when we adding another node, it is 3 >= 3, we should use the strict mode because one node is losing the range, we should stream from the node losing the range.